node/perf-test/src/command.rs

```diff
@@ -0,0 +1,415 @@
+use sp_keystore::{testing::KeyStore, KeystoreExt, SyncCryptoStorePtr};
+use sp_api::{ConstructRuntimeApi, ProvideRuntimeApi, BlockId};
+use std::{fmt::Debug, sync::Arc, marker::PhantomData, time};
+use sp_state_machine::StateMachine;
+use cli_opt::RpcConfig;
+use fp_rpc::EthereumRuntimeRPCApi;
+use nimbus_primitives::NimbusId;
+use cumulus_primitives_parachain_inherent::{
+	MockValidationDataInherentDataProvider, ParachainInherentData,
+};
+use sc_consensus_manual_seal::{run_manual_seal, EngineCommand, ManualSealParams};
+use ethereum::{TransactionAction, TransactionSignature};
+
+use async_io::Timer;
+use futures::{Stream, Sink, SinkExt, channel::mpsc::Sender};
+
+use service::{chain_spec, RuntimeApiCollection, Block, TransactionConverters};
+
+type FullClient<RuntimeApi, Executor> = TFullClient<Block, RuntimeApi, Executor>;
+type FullBackend = TFullBackend<Block>;
+
+struct PerfTestRunner<RuntimeApi, Executor>
+	where
+		RuntimeApi:
+			ConstructRuntimeApi<Block, FullClient<RuntimeApi, Executor>> + Send + Sync + 'static,
+		RuntimeApi::RuntimeApi:
+			RuntimeApiCollection<StateBackend = sc_client_api::StateBackendFor<FullBackend, Block>>,
+		Executor: NativeExecutionDispatch + 'static,
+{
+	task_manager: TaskManager,
+	client: Arc<TFullClient<Block, RuntimeApi, Executor>>,
+	manual_seal_command_sink: futures::channel::mpsc::Sender<EngineCommand<H256>>,
+	pool: sc_transaction_pool::FullPool<Block, FullClient<RuntimeApi, Executor>>,
+
+	_marker1: PhantomData<RuntimeApi>,
+	_marker2: PhantomData<Executor>,
+}
+
+// TODO: am I abusing the name "runner"?
+impl<RuntimeApi, Executor> PerfTestRunner<RuntimeApi, Executor>
+	where
+		RuntimeApi:
+			ConstructRuntimeApi<Block, FullClient<RuntimeApi, Executor>> + Send + Sync + 'static,
+		RuntimeApi::RuntimeApi:
+			RuntimeApiCollection<StateBackend = sc_client_api::StateBackendFor<FullBackend, Block>>,
+		Executor: NativeExecutionDispatch + 'static,
+{
+	fn from_cmd(config: &Configuration, cmd: &PerfCmd) -> CliResult<Self> {
+
+		// TODO: can we explicitly disable everything RPC-related? or anything network related, for
+		//       that matter?
+		//       Note: new_dev() calls `network_starter.start_network()` at the very end
+		let rpc_config = RpcConfig {
+			ethapi: Default::default(),
+			ethapi_max_permits: 0,
+			ethapi_trace_max_count: 0,
+			ethapi_trace_cache_duration: 0,
+			max_past_logs: 0,
+		};
+
+		// TODO: allow CLI to configure or at least make sure this doesn't conflict with available
+		//       CLI options
+		let sealing = cli_opt::Sealing::Manual;
+
+		let sc_service::PartialComponents {
+			client,
+			backend,
+			mut task_manager,
+			import_queue,
+			keystore_container,
+			select_chain: maybe_select_chain,
+			transaction_pool,
+			other:
+				(
+					block_import,
+					pending_transactions,
+					filter_pool,
+					telemetry,
+					_telemetry_worker_handle,
+					frontier_backend,
+				),
+		} = service::new_partial::<RuntimeApi, Executor>(&config, true)?;
```

Comment:
I believe if you are using the dev service you will not be creating a PoV. That might be fine here, but worth keeping in mind.